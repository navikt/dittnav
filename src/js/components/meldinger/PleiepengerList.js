import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { FormattedMessage as F } from 'react-intl';

class PleiepengerList extends Component {
  render() {
    if (!this.props.pleiepenger || this.props.pleiepenger.length === 0) return null;

    return (
      <div className="message">
        <span className="icon pleiepenger-icon" aria-label="pleiepenger-ikon" />
        <div className="texts">
          {this.props.pleiepenger.map(l => (
            <p key={l.child + l.usedPerToday} ><F id="pleiepenger.peridag" values={{ child: l.child }} /> <b>{l.usedPerToday}</b></p>
            ))}
        </div>
      </div>
    );
  }
}

export const Pleiepenger = PropTypes.shape({ child: PropTypes.string, usedPerToday: PropTypes.string });

PleiepengerList.propTypes = {
  pleiepenger: PropTypes.arrayOf(Pleiepenger),
};

PleiepengerList.defaultProps = {
  pleiepenger: null,
};

export default PleiepengerList;
