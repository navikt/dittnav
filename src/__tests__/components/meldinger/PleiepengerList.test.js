import * as React from 'react';
const ReactTestRenderer = require('react-test-renderer');
import PleiepengerList from 'js/components/meldinger/PleiepengerList';

test('empty PleiepengerList green test', () => {
  const component = ReactTestRenderer.create((<PleiepengerList />));
  expect(component.toJSON()).toMatchSnapshot();
});

test('PleiepengerList empty list', () => {
  const component = ReactTestRenderer.create((<PleiepengerList pleiepenger={[]} />));
  expect(component.toJSON()).toMatchSnapshot();
});

test('PleiepengerList list with one element', () => {
  const component = ReactTestRenderer.create((<PleiepengerList pleiepenger={[{child: "123321", usedPerToday: "123"}]} />));
  expect(component.toJSON()).toMatchSnapshot();
});

test('PleiepengerList list with two elements', () => {
  const component = ReactTestRenderer.create((<PleiepengerList pleiepenger={[{child: "123321", usedPerToday: "123"}, {child: "123321", usedPerToday: "12"}]} />));
  expect(component.toJSON()).toMatchSnapshot();
});