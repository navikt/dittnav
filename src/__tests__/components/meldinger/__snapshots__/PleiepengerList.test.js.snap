// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PleiepengerList empty list 1`] = `
<div
  className="message"
>
  <span
    aria-label="pleiepenger-ikon"
    className="icon pleiepenger-icon"
  />
  <div
    className="texts"
  />
</div>
`;

exports[`PleiepengerList list with one element 1`] = `
<div
  className="message"
>
  <span
    aria-label="pleiepenger-ikon"
    className="icon pleiepenger-icon"
  />
  <div
    className="texts"
  >
    <p>
      Forbrukte pleiepengedager per i dag for barn med f.nr. 
      123321
      : 
      <b>
        123
      </b>
    </p>
  </div>
</div>
`;

exports[`PleiepengerList list with two elements 1`] = `
<div
  className="message"
>
  <span
    aria-label="pleiepenger-ikon"
    className="icon pleiepenger-icon"
  />
  <div
    className="texts"
  >
    <p>
      Forbrukte pleiepengedager per i dag for barn med f.nr. 
      123321
      : 
      <b>
        123
      </b>
    </p>
    <p>
      Forbrukte pleiepengedager per i dag for barn med f.nr. 
      123321
      : 
      <b>
        12
      </b>
    </p>
  </div>
</div>
`;

exports[`empty PleiepengerList green test 1`] = `null`;
